api/clients/eigenda_client.go

```diff
@@ -298,6 +304,125 @@ func (m *EigenDAClient) putBlob(ctx context.Context, rawData []byte, resultChan
 	}
 }
 
+// PaidPutBlob behaves like PutBlob but with authenticated payment.
```

Comment:
yep, did a similar refactoring that we did on the server side; if there's a setting on "PaymentSignerPrivateKeyHex", then we call out to `DispersePaidBlob`, otherwise `DisperseBlobAuthenticated`. lmk if this is okay?